datarobot_provider/operators/autopilot.py

```diff
@@ -0,0 +1,115 @@
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+from typing import Any
+from typing import Dict
+from typing import Iterable
+
+import datarobot as dr
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+
+from datarobot_provider.hooks.datarobot import DataRobotHook
+
+DATAROBOT_MAX_WAIT = 600
+
+
+class StartAutopilotOperator(BaseOperator):
+    """
+    Triggers DataRobot Autopilot to train set of models.
+
+    :param project_id: DataRobot project ID
+    :type project_id: str
+    :param featurelist_id: Specifies which feature list to use.
+    :type featurelist_id: str, optional
+    :param relationships_configuration_id: ID of the relationships configuration to use.
+    :type relationships_configuration_id: str, optional
+    :param segmentation_task_id: The segmentation task that should be used to split the project
+            for segmented modeling.
+    :type segmentation_task_id: str, optional
+    :param max_wait_sec: For some settings, an asynchronous task must be run to analyze the dataset.  max_wait
+            governs the maximum time (in seconds) to wait before giving up.
+    :type max_wait_sec: int, optional
+    :param datarobot_conn_id: Connection ID, defaults to `datarobot_default`
+    :type datarobot_conn_id: str, optional
+    """
+
+    # Specify the arguments that are allowed to parse with jinja templating
+    template_fields: Iterable[str] = [
+        "project_id",
+        "featurelist_id",
+        "relationships_configuration_id",
+        "segmentation_task_id",
+    ]
+    template_fields_renderers: Dict[str, str] = {}
+    template_ext: Iterable[str] = ()
+    ui_color = '#f4a460'
+
+    def __init__(
+        self,
+        *,
+        project_id: str,
+        featurelist_id: str = None,
+        relationships_configuration_id: str = None,
+        segmentation_task_id: str = None,
+        max_wait_sec: int = DATAROBOT_MAX_WAIT,
+        datarobot_conn_id: str = "datarobot_default",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.project_id = project_id
+        self.featurelist_id = featurelist_id
+        self.relationships_configuration_id = relationships_configuration_id
+        self.segmentation_task_id = segmentation_task_id
+        self.max_wait_sec = max_wait_sec
+        self.datarobot_conn_id = datarobot_conn_id
+
+        if kwargs.get('xcom_push') is not None:
+            raise AirflowException(
+                "'xcom_push' was deprecated, use 'BaseOperator.do_xcom_push' instead"
+            )
+
+    def execute(self, context: Dict[str, Any]) -> None:
+        # Initialize DataRobot client
+        DataRobotHook(datarobot_conn_id=self.datarobot_conn_id).run()
+        # Train models
+        project = dr.Project.get(self.project_id)
+        if project.target:
+            self.log.info(f"Models are already trained for project_id={project.id}")
```

Comment:
Should this be at least a warning or even an error? Is it a valid usecase for a user to provide a project that is already trained?